src/Core/Authentication.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * @file /src/Core/Authentication.php
+ */
+
+namespace Friendica\Core;
+
+use Friendica\Core\Addon;
+use Friendica\BaseObject;
+use Friendica\Core\Config;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Database\DBA;
+use Friendica\Core\L10n;
+use Friendica\Core\PConfig;
+
+/**
+* Handle Authentification, Session and Cookies
+*/    
+class Authentication extends BaseObject
+{
+	/**
+	 * @brief Calculate the hash that is needed for the "Friendica" cookie
+	 *
+	 * @param array $user Record from "user" table
+	 *
+	 * @return string Hashed data
+	 */
+	public static function cookie_hash($user)
+	{
+		return(hash("sha256", Config::get("system", "site_prvkey") .
+				$user["prvkey"] .
+				$user["password"]));
+	}
+
+	/**
+	 * @brief Set the "Friendica" cookie
+	 *
+	 * @param int $time
+	 * @param array $user Record from "user" table
+	 */
+	public static  function new_cookie($time, $user = [])
+	{
+		if ($time != 0) {
+			$time = $time + time();
+		}
+
+		if ($user) {
+			$value = json_encode(["uid" => $user["uid"],
+				"hash" => self::cookie_hash($user),
+				"ip" => defaults($_SERVER, 'REMOTE_ADDR', '0.0.0.0')]);
```

Comment:
Watch out for global functions, their corresponding file should be `require_once` by this file until we move them to classes.